src/js/player.js

```diff
@@ -3444,6 +3444,18 @@ class Player extends Component {
     this.on('mousemove', handleMouseMove);
     this.on('mouseup', handleMouseUp);
 
+    this.on(this.getChild('controlBar'), 'mouseenter', function(event) {
```

Comment:
we should store the control bar object in a variable (to also re-use below) and do a null-check on it.

Additionally, it would be better to call `.on` directly on the contrlBar itself as the handler will automatically be removed when the control bar is disposed. I don't think the current event handler gets cleared up automatically, currently.

Reply:
I'm sorry, I didn't really understand your second point. Wouldn't it be the same if I associated the handler on the immediate return of the getChild function or a variable with its value? Or do you mean something else?

Reply:
Basically, `this.on(anotherComponent, 'mouseenter', () => {})` vs `anotherComponent.on('mouseenter', () => {})`. I am unsure that the first one will have the `mouseenter` event removed when the component is disposed where-as I'm certain the second one will be done properly.

Ultimately, the code should end up something like:
```js
const controlBar = this.getChild('controlBar');

if (controlBar) {
  controlBar.on('mouseenter', (event) => /* ... */);
  controlBar.on('mouseleave', (event) => /* ... */);
}
```

Reply:
I completely understand what you say so now we can see that on the code.